srtcore/core.cpp

```diff
@@ -3096,9 +3096,138 @@ bool CUDT::checkApplyFilterConfig(const std::string &confstr)
+    // // That is, given socket is the first one.
+    // The group data should be set up with its own data. They should already be passed here
+    // in the variables.
+    //
+    // Override the schedule sequence of the group in this case because whatever is set now,
+    // it's not valid.
+
+    HLOGC(dlog.Debug, log << "applyGroupSequences: no socket found connected and transmitting, @"
+            << target << " not changing sequences, storing snd-seq %" << (w_snd_isn));
+
+    currentSchedSequence(w_snd_isn);
+
+    return true;
+}
+
+bool CUDTGroup::getMasterData(SRTSOCKET slave, SRTSOCKET& w_mpeer, steady_clock::time_point& w_st)
+{
+    // Find at least one connection, which is running. Note that this function is called
+    // from within a handshake process, so the socket that undergoes this process is at best
+    // currently in GST_PENDING state and it's going to be in GST_IDLE state at the
+    // time when the connection process is done, until the first reading/writing happens.
+    CGuard cg (m_GroupLock);
+
+    for (gli_t gi = m_Group.begin(); gi != m_Group.end(); ++gi)
+    {
+        if (gi->sndstate == GST_RUNNING)
+        {
+            // Found it. Get the socket's peer's ID and this socket's
+            // Start Time. Once it's delivered, this can be used to calculate
+            // the Master-to-Slave start time difference.
+            w_mpeer = gi->ps->m_PeerID;
+            w_st = gi->ps->core().socketStartTime();
+            HLOGC(mglog.Debug, log << "getMasterData: found RUNNING master @" << gi->id
+                << " - reporting master's peer $" << w_mpeer << " starting at "
+                << FormatTime(w_st));
+            return true;
+        }
+    }
+
+    // If no running one found, then take the first socket in any other
+    // state than broken, except the slave. This is for a case when a user
+    // has prepared one link already, but hasn't sent anything through it yet.
+    for (gli_t gi = m_Group.begin(); gi != m_Group.end(); ++gi)
+    {
+        if (gi->sndstate == GST_BROKEN)
+            continue;
+
+        if (gi->id == slave)
+            continue;
+
+        // Found it. Get the socket's peer's ID and this socket's
+        // Start Time. Once it's delivered, this can be used to calculate
+        // the Master-to-Slave start time difference.
+        w_mpeer = gi->ps->core().m_PeerID;
+        w_st = gi->ps->core().socketStartTime();
+        HLOGC(mglog.Debug, log << "getMasterData: found IDLE/PENDING master @" << gi->id
+            << " - reporting master's peer $" << w_mpeer << " starting at "
+            << FormatTime(w_st));
+        return true;
+    }
+
+    HLOGC(mglog.Debug, log << "getMasterData: no link found suitable as master for @" << slave);
+    return false;
+}
+
 void CUDT::startConnect(const sockaddr_any& serv_addr, int32_t forced_isn)
 {
-    CGuard cg(m_ConnectionLock);
+    CGuard cg (m_ConnectionLock);
 
```

Comment:
extra whitespace?

Reply:
Intended. Though it will get disappeared probably after the reformatting action.